### PR TITLE
Fix flaky build id issue

### PR DESCRIPTION
**Pull request by Sushisource**
This should fix the only case where I could imagine the id not being set. Could not repro locally.

### CODE REVIEW

File: internal/internal_task_handlers.go [hunk 1]
```diff
@@ -1028,10 +1028,16 @@ ProcessEvents:
 		}
 		if isReplay {
 			w.workflowInfo.currentTaskBuildID = currentBuildID
-		} else if !isReplay && !isQueryOnlyTask {
+		} else if !isQueryOnlyTask {
 			// Query only tasks should use the build ID from the workflow task, not the worker's
```
Comment (Quinn-With-Two-Ns) [inline internal/internal_task_handlers.go:1032]: This comment seem irrelevant now?

File: internal/internal_task_handlers.go [hunk 2]
```diff
@@ -1028,10 +1028,16 @@ ProcessEvents:
...
 			// build id, since the user cares about what affected workflow state.
 			w.workflowInfo.currentTaskBuildID = w.wth.workerBuildID
+		} else {
+			// It is a query only task, and we're not replaying somehow - still use the task's build
```
Comment (Quinn-With-Two-Ns) [inline internal/internal_task_handlers.go:1036]: I   don't  understand this comment, didn't we discuss before that queries are always treated as not replaying?

Comment (Sushisource) [reply internal/internal_task_handlers.go:1036]: Yeah, I can just get rid of the not replaying part

File: internal/internal_task_handlers.go (hunk 1 above)
Comment (Sushisource) [reply internal/internal_task_handlers.go]: I made them better

File: internal/internal_task_handlers.go [hunk 3]
```diff
@@ -1028,10 +1028,15 @@ ProcessEvents:
...
-		} else if !isReplay && !isQueryOnlyTask {
-			// Query only tasks should use the build ID from the workflow task, not the worker's
-			// build id, since the user cares about what affected workflow state.
+		} else if !isQueryOnlyTask {
+			// Real new (non-query, non-replay) tasks use the worker's build ID
 			w.workflowInfo.currentTaskBuildID = w.wth.workerBuildID
+		} else {
+			// It *is* a query only task, use the task's build ID (queries are considered non-replay
+			// in go sdk), but only if one was actually discovered.
+			if currentBuildID != "" {
```
Comment (Quinn-With-Two-Ns) [inline internal/internal_task_handlers.go:1037]: When is `currentBuildID` empty?

Comment (Sushisource) [reply internal/internal_task_handlers.go:1037]: It can be empty here if the lookahead didn't find anything, but we wouldn't want to overwrite some previous task which did find something.

Comment (Quinn-With-Two-Ns) [reply internal/internal_task_handlers.go:1037]: Yeah I am wondering what scenario look ahead can't find anything?

Comment (Sushisource) [reply internal/internal_task_handlers.go:1037]: The problem is sort of specific to queries on the first task I think. No WFT completed event exists yet to be looked-ahead-to (well, it does actually in the query, but we don't read it since we're already up-to-date), but the `nextTask` code still runs, which then returns this as empty.

Comment (Quinn-With-Two-Ns) [reply internal/internal_task_handlers.go:1037]: Ah I didn't think of the first WFT thanks for clarifying

**Review by Quinn-With-Two-Ns — APPROVED** [review]
